types/index.ts

```diff
@@ -0,0 +1,37 @@
+export type ButtonProps = {
+  className: string;
+  children: any;
+  color: string;
+  icon: any;
+  leftIcon: any;
+  onClick: any;
+  outline: boolean;
+  rightIcon: any;
+  size: string;
+};
+
+export type Sizes = {
+  xs: string;
+  sm: string;
+  md: string;
+  lg: string;
+  xl: string;
+};
+
+export type Styles = {
+  solid: {
+    primary: string;
```

Comment:
Lets define the color styles in a variable and use the variable everywhere it is needed. The same variable can be used for `solid` on line 22 and `outline` on line 30.